portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,114 @@
+      'thumb position. This double-jointedness actually makes it a bit ' +
+      'difficult to play the cello, as collapsing joints are a big no-no ' + 
+      'for your left hand.'));
+  const post2 = [p2];
+  posts = [post1, post2];
+});
+
 /**
- * Adds a random greeting to the page.
+ * Adds a random fact to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function addRandomFact() {
+  const facts =
+      ['I love hot cheetos.', 'Purple is my favorite color.', 
+      'I\'m horrible at remembering song lyrics.', 'I speak some French.'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  // Pick a random fact.
+  const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  const factContainer = document.getElementById('fact-container');
+  factContainer.innerText = fact;
+}
+
+/**
+ * Shows the specified blog post to the user
+ * @param {number} postNum The number of which
+ * blog post to show.
+ */
+function showBlogPost(postNum) {
+  // Find the correct triangle to expand and switch its image.
+  // Also change the onclick function for this button to hide the post.
+  const buttonId = "expandTriangle" + postNum;
+  let button = document.getElementById(buttonId);
+  console.log(button);
+  if (button !== null) {
+    button.src = "/images/downtriangle.jpeg";
+    button.onclick = function() {
+      hideBlogPost(postNum);
+    };
+  }
+
+  // Add blog post content to the correct area.
+  const blogPostId = "blogPostArea" + postNum;
+  let postArea = document.getElementById(blogPostId);
+  if (postArea !== null) {
+    if (postNum > 0 && postNum <= posts.length) {
+      const elements = posts[postNum-1]
+      for (i = 0; i < elements.length; i++) {
+        const element = elements[i]
+        if (element !== null) {
+           postArea.appendChild(element);
+        }
+      }
+    }
+  }
+
+  // Automatically scroll window with post now opened.
+  window.scrollBy(0, 500);
+}
+
+/**
+ * Hides the specified blog post from the user
+ * @param {number} postNum The number of which
+ * blog post to hide.
+ */
+function hideBlogPost(postNum) {
+  // Find the correct triangle to hide and switch its image.
+  // Also change the onclick function for this button to show the post.
+  const buttonId = "expandTriangle" + postNum;
+  let button = document.getElementById(buttonId);
+  console.log(button);
+  if (button !== null) {
+    button.src = "/images/righttriangle.jpeg";
+    button.onclick = function() {
+      showBlogPost(postNum);
+    };
+  }  
+
+  // Remove blog post content from its area.
+  const blogPostId = "blogPostArea" + postNum;
+  let postArea = document.getElementById(blogPostId);
+  if (postArea !== null) {
```

Comment:
If postArea is null, is this considered an error? If so, consider logging the error.